mmv1/products/networksecurity/SecurityProfile.yaml

```diff
@@ -0,0 +1,176 @@
+      and may be sent on update and delete requests to ensure the client has an up-to-date
+      value before proceeding.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of the security profile. The Max length is 512 characters.
+  - !ruby/object:Api::Type::KeyValueLabels
+    name: 'labels'
+    description: |
+      A map of key/value label pairs to assign to the resource.
+  - !ruby/object:Api::Type::NestedObject
+    name: threatPreventionProfile
+    description: The threat prevention configuration for the security profile.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: severityOverrides
+        description: The configuration for overriding threats actions by severity match.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::Enum
+              name: 'action'
+              description: Threat action override.
+              required: true
+              values:
+                - :ALERT
+                - :ALLOW
+                - :DEFAULT_ACTION
+                - :DENY
+                - :THREAT_ACTION_UNSPECIFIED
+            - !ruby/object:Api::Type::Enum
+              name: 'severity'
+              description: Severity level to match.
+              required: true
+              values:
+                - :CRITICAL
+                - :HIGH
+                - :INFORMATIONAL
+                - :LOW
+                - :MEDIUM
+                - :SEVERITY_UNSPECIFIED
+      - !ruby/object:Api::Type::Array
+        name: threatOverrides
+        description: |
+          The configuration for overriding threats actions by threat id match.
+          If a threat is matched both by configuration provided in severity overrides
+          and threat overrides, the threat overrides action is applied.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::Enum
+              name: 'action'
+              description: Threat action.
+              required: true
+              values:
+                - :ALERT
+                - :ALLOW
+                - :DEFAULT_ACTION
+                - :DENY
+                - :THREAT_ACTION_UNSPECIFIED
+            - !ruby/object:Api::Type::String
+              name: 'threatId'
+              description: Vendor-specific ID of a threat to override.
+              required: true
+            - !ruby/object:Api::Type::Enum
+              name: 'type'
+              description: Type of threat.
+              output: true
+              values:
+                - :ANTIVIRUS
+                - :DNS
+                - :THREAT_TYPE_UNSPECIFIED
+                - :UNKNOWN
+                - :VULNERABILITY
+                - :SPYWARE
+  - !ruby/object:Api::Type::Enum
+    name: 'type'
+    description: The type of security profile.
+    immutable: true
+    default_value: :THREAT_PREVENTION
```

Comment:
I would consider not using a default value here, because defaults are difficult to change or remove later (they require a breaking change to the provider). Seeing this only has one value right now also implies to me that this might be expected to change over time. However, if you're confident the default will make sense long-term it is probably fine.

Reply:
Makes sense. Removed the default and made it required.